### PR TITLE
Clean up Container Create

### PR DESCRIPTION
**Pull request by anthony-murphy**

### CODE REVIEW

File: packages/loader/container-loader/src/container.ts [hunk 1]
```diff
@@ -238,20 +230,32 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
...
-        if (source.create) {
-            await container.createDetached(source.codeDetails);
-        } else {
-            await container.rehydrateDetachedFromSnapshot(source.snapshot);
-        }
-
+    /**
+     * Create a new container in a detached that is initialize with a
+     * snapshot from a previous detached container.
+     */
```
Comment (jatgarg) [inline packages/loader/container-loader/src/container.ts:250]: nit: detached "state" and initialize -> "initialized"

**Review by jatgarg — APPROVED** [review]
:shipit:

**Review by ChumpChief — APPROVED** [review]

File: packages/loader/container-loader/src/container.ts (hunk 1 above)
Comment (anthony-murphy) [reply packages/loader/container-loader/src/container.ts:250]: please us to suggest feature to just fix stuff like this :)

File: packages/loader/container-loader/src/container.ts [hunk 2]
```diff
@@ -238,20 +230,32 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
...
+    }
 
-        if (source.create) {
-            await container.createDetached(source.codeDetails);
-        } else {
-            await container.rehydrateDetachedFromSnapshot(source.snapshot);
-        }
-
+    /**
+     * Create a new container in a detached that is initialize with a
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/container.ts:248]:
```suggestion
     * Create a new container in a detached state that is initialized with a
```

**Pull request merged**